rocsolver/library/src/rocsolver_logger.cpp

```diff
@@ -150,12 +182,38 @@ rocblas_status rocsolver_logging_cleanup()
         *logger->profile_os << std::endl;
     }
 
-    delete logger;
-    logger = nullptr;
+    // reset to no logging and clear profile info
+    logger->max_levels = 1;
+    logger->layer_mode = rocblas_layer_mode_none;
```

Comment:
It just occurred to me that the values provided using environment variables will only be used upon first creating the logger. Should we default back to these values when doing the cleanup?

Reply:
I rather go back to the non-logging default. This way cleanup is also used as a way to suspend logging which can be restarted with the loggin_initialize. As the static logger object is not deleted we can even pass an option to clean or not the profile unordered map; this way a user can suspend profiling in a section of its application code, and re-activate it again later...